api/v1alpha1/etcdcluster_webhook.go

```diff
@@ -66,6 +67,23 @@ func (r *EtcdCluster) Default() {
 			}
 		}
 	}
+	etcdContainerIdx := slices.IndexFunc(r.Spec.PodTemplate.Spec.Containers, func(c corev1.Container) bool {
+		return c.Name == "etcd"
+	})
+	if etcdContainerIdx == -1 {
+		r.Spec.PodTemplate.Spec.Containers = append(r.Spec.PodTemplate.Spec.Containers, corev1.Container{
+			Name:  "etcd",
+			Image: defaultEtcdImage,
+		})
+	} else {
+		c := &r.Spec.PodTemplate.Spec.Containers[etcdContainerIdx]
+		if c.Image == "" {
+			c.Image = defaultEtcdImage
+		}
+		if c.ImagePullPolicy == "" {
+			c.ImagePullPolicy = corev1.PullIfNotPresent
```

Comment:
Afaik we can leave it just unspecified. K8s will automatically set IfNotPresent as default value if not specified in container

Reply:
You're right. Thank you!